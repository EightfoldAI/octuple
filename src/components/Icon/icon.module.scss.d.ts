declare namespace IconModuleScssModule {
    export interface IIconModuleScss {
        body1: string;
        body2: string;
        body3: string;
        'button-spacer-1': string;
        'button-spacer-2': string;
        'button-spacer-3': string;
        button1: string;
        button2: string;
        button3: string;
        buttonSpacer1: string;
        buttonSpacer2: string;
        buttonSpacer3: string;
        caption: string;
        'chevron-icon': string;
        'chevron-spacer': string;
        chevronIcon: string;
        chevronSpacer: string;
        'custom-padding-vertical': string;
        customPaddingVertical: string;
        display1: string;
        display2: string;
        display3: string;
        display4: string;
        'divider-line': string;
        dividerLine: string;
        'flex-structure-horizontal': string;
        'flex-structure-vertical': string;
        flexStructureHorizontal: string;
        flexStructureVertical: string;
        'full-height': string;
        fullHeight: string;
        header1: string;
        header2: string;
        header3: string;
        header4: string;
        header5: string;
        header6: string;
        icon: string;
        'icon-1-material': string;
        'icon-2-material': string;
        'icon-3-material': string;
        'icon-4-material': string;
        'icon-material-l': string;
        'icon-material-m': string;
        'icon-material-s': string;
        'icon-material-xs': string;
        icon1Material: string;
        icon2Material: string;
        icon3Material: string;
        icon4Material: string;
        iconMaterialL: string;
        iconMaterialM: string;
        iconMaterialS: string;
        iconMaterialXs: string;
        iconWrapper: string;
        l: string;
        'logo-text': string;
        logoText: string;
        m: string;
        mappings: string;
        names: string;
        'octuple-menu': string;
        octupleMenu: string;
        'open-sans': string;
        openSans: string;
        'pre-display': string;
        preDisplay: string;
        s: string;
        semibold: string;
        sourceRoot: string;
        sources: string;
        sourcesContent: string;
        'space-m': string;
        spaceM: string;
        spacer: string;
        subtitle1: string;
        subtitle2: string;
        version: string;
        xl: string;
        xxl: string;
        xxxl: string;
    }
}

declare const IconModuleScssModule: IconModuleScssModule.IIconModuleScss & {
    /** WARNING: Only available when `css-loader` is used without `style-loader` or `mini-css-extract-plugin` */
    locals: IconModuleScssModule.IIconModuleScss;
};

export = IconModuleScssModule;
