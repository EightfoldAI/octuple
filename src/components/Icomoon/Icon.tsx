import React, { FC, useContext } from 'react';
import { mergeClasses } from '../../shared/utilities';
import { IcomoonProps, IcomoonSize } from './';
import IcomoonReact from 'icomoon-react';

import styles from './icon.module.scss';

const IconSetContext = React.createContext<object | null>(null);

export const IcomoonProvider = IconSetContext.Provider;

export const Icomoon: FC<IcomoonProps> = ({
    ariaHidden = false,
    classNames,
    color,
    iconName,
    id,
    role = 'presentation',
    size = IcomoonSize.Medium,
    'data-test-id': dataTestId,
}) => {
    const iconClassNames: string = mergeClasses([
        classNames,
        styles.iconWrapper,
    ]);

    return (
        <span
            data-test-id={dataTestId}
            aria-hidden={ariaHidden}
            className={iconClassNames}
            id={id}
            role={role}
        >
            <IconSetContext.Consumer>
                {(iconSet) => (
                    <IcomoonReact
                        iconSet={iconSet}
                        icon={iconName}
                        size={size}
                        color={color}
                    />
                )}
            </IconSetContext.Consumer>
        </span>
    );
};
