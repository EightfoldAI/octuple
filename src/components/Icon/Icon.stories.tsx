import React from 'react';
import { Icon, IconName, IconSize } from './index';

export default {
    title: 'Icon',
    component: Icon,
};

export const Icons = () => (
    <>
        <h2>Icons</h2>
        <p>X-Small</p>
        <Icon path={IconName.mdiCardsHeart} size={IconSize.XSmall} />
        <p>Small</p>
        <Icon path={IconName.mdiCardsHeart} size={IconSize.Small} />
        <p>Medium</p>
        <Icon path={IconName.mdiCardsHeart} />
        <p>Large</p>
        <Icon path={IconName.mdiCardsHeart} size={IconSize.Large} />
    </>
);
