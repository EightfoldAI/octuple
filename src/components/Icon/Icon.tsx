'use client';

import React, { FC } from 'react';
import IcomoonReact from 'icomoon-react';

import { mergeClasses } from '../../shared/utilities';
import { Icon as MdiIcon } from '@mdi/react';
import { IconProps, IconSize } from './index';

import styles from './icon.module.scss';
import { useConfig } from '../ConfigProvider';

export const Icon: FC<IconProps> = ({
  ariaHidden = false,
  classNames,
  color,
  description,
  horizontal,
  id,
  path,
  role = 'presentation',
  rotate,
  size = IconSize.Medium,
  spin,
  style,
  title,
  vertical,
  'data-test-id': dataTestId,
  icomoonIconName,
}) => {
  const { icomoonIconSet } = useConfig();

  const iconClassNames: string = mergeClasses([classNames, styles.iconWrapper]);

  if (!icomoonIconName && !path) {
    return null;
  }

  const iconComponent = icomoonIconName ? (
    <IcomoonReact
      iconSet={icomoonIconSet}
      size={size}
      color={color}
      icon={icomoonIconName}
      role="presentation"
    />
  ) : (
    <MdiIcon
      color={color}
      description={description}
      horizontal={horizontal}
      path={path}
      rotate={rotate}
      size={size}
      title={title}
      vertical={vertical}
      spin={spin}
    />
  );

  return (
    <span
      data-test-id={dataTestId}
      aria-hidden={ariaHidden}
      className={iconClassNames}
      id={id}
      role={role}
      style={style ? style : null}
    >
      {iconComponent}
    </span>
  );
};

Icon.displayName = 'Icon';
