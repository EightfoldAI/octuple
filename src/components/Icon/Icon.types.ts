import { IconName } from './mdi';
import { IconProps as MdiIconProps } from '@mdi/react/dist/IconProps';

export enum IconSize {
    Large = '1.5rem',
    Medium = '1.25rem',
    Small = '1rem',
    XSmall = '0.875rem',
}

export interface IconProps extends MdiIconProps {
    /**
     * The icon svg path name.
     */
    path: IconName;
    /**
     * The icon is aria-hidden.
     * @default false
     */
    ariaHidden?: boolean;
    /**
     * The icon class names.
     */
    className?: string;
    /**
     * The icon color.
     */
    color?: string;
    /**
     * The icon description.
     */
    description?: string;
    /**
     * The icon is horizontal.
     * @default false
     */
    horizontal?: boolean;
    /**
     * The icon id.
     */
    id?: string;
    /**
     * The icon aria role.
     * @default 'presentation'
     */
    role?: string;
    /**
     * The icon is rotated.
     * @default 0
     */
    rotate?: number;
    /**
     * The icon size.
     * @default IconSize.Medium
     */
    size?: IconSize;
    /**
     * The icon spin animation timer in milliseconds, seconds.
     * @default false
     */
    spin?: boolean | number;
    /**
     * The icon title.
     */
    title?: string;
    /**
     * The icon is vertical.
     * @default false
     */
    vertical?: boolean;
}
