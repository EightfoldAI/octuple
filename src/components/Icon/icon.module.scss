.iconWrapper {
    display: inline-block;
    margin: 0;
    padding: 0;

    svg {
        vertical-align: middle;
    }
}