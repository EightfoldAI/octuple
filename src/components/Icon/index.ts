export * from './Icon';
export * from './Icon.types';
export * from './mdi';
