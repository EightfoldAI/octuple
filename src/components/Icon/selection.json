{
    "IcoMoonType": "selection",
    "icons": [
        {
            "icon": {
                "paths": [
                    "M102.4 153.6v716.8h241.912v-139.999h79.375v139.999h241.912v-716.8h-563.2zM181.777 233.867h81.268v78.399h-81.268v-78.399zM344.312 233.867h79.375v78.399h-79.375v-78.399zM504.955 233.867h81.267v78.399h-81.267v-78.399zM181.777 392.532h81.268v80.269h-81.268v-80.269zM344.312 392.532h79.375v80.269h-79.375v-80.269zM504.955 392.532h81.267v80.269h-81.267v-80.269zM181.777 551.201h81.268v80.266h-81.268v-80.266zM344.312 551.201h79.375v80.266h-79.375v-80.266zM504.955 551.201h81.267v80.266h-81.267v-80.266zM181.777 711.734h81.268v78.4h-81.268v-78.4zM504.955 711.734h81.267v78.4h-81.267v-78.4z",
                    "M598.536 358.4h108.529c14.303 0 26.501 5.478 36.598 16.435 10.936 10.112 16.404 22.333 16.404 36.662v54.359h108.531c14.303 0 26.501 5.478 36.598 16.435 10.936 10.115 16.404 22.756 16.404 37.926v295.821c0 15.171-5.468 28.234-16.404 39.191-10.097 10.115-22.295 15.171-36.598 15.171h-431.596c-14.303 0-26.921-5.056-37.86-15.171-10.094-10.957-15.142-24.020-15.142-39.191v-295.821c0-15.171 5.048-27.812 15.142-37.926 10.939-10.957 23.557-16.435 37.86-16.435h108.531v-54.359c0-14.328 5.046-26.55 15.142-36.662 10.936-10.957 23.557-16.435 37.86-16.435zM707.064 465.856v-54.359h-108.529v54.359h108.529z"
                ],
                "attrs": [
                    { "fill": "rgb(80, 206, 225)" },
                    { "fill": "rgb(20, 109, 166)", "opacity": 0.6 }
                ],
                "isMulticolor": true,
                "isMulticolor2": true,
                "grid": 0,
                "tags": ["CareerHub"],
                "colorPermutations": {
                    "201091661802062251": [{ "f": 1 }, { "f": 0 }]
                }
            },
            "attrs": [
                { "fill": "rgb(80, 206, 225)" },
                { "fill": "rgb(20, 109, 166)", "opacity": 0.6 }
            ],
            "properties": {
                "order": 3,
                "id": 0,
                "name": "CareerHub",
                "prevSize": 32,
                "code": 59648,
                "codes": [59648, 59649]
            },
            "setIdx": 0,
            "setId": 2,
            "iconIdx": 0
        },
        {
            "icon": {
                "paths": [
                    "M1024 608l-192-192v-288h-128v160l-192-192-512 512v32h128v320h320v-192h128v192h320v-320h128z"
                ],
                "tags": ["home", "house"],
                "defaultCode": 59650,
                "grid": 16,
                "attrs": []
            },
            "attrs": [],
            "properties": {
                "ligatures": "home3, house3",
                "name": "home3",
                "order": 4,
                "id": 3,
                "prevSize": 32,
                "code": 59650
            },
            "setIdx": 1,
            "setId": 1,
            "iconIdx": 2
        },
        {
            "icon": {
                "paths": [
                    "M0 1024h512v-1024h-512v1024zM320 128h128v128h-128v-128zM320 384h128v128h-128v-128zM320 640h128v128h-128v-128zM64 128h128v128h-128v-128zM64 384h128v128h-128v-128zM64 640h128v128h-128v-128zM576 320h448v64h-448zM576 1024h128v-256h192v256h128v-576h-448z"
                ],
                "tags": ["office", "buildings", "work"],
                "defaultCode": 59651,
                "grid": 16,
                "attrs": []
            },
            "attrs": [],
            "properties": {
                "ligatures": "office, buildings",
                "name": "office",
                "order": 5,
                "id": 4,
                "prevSize": 32,
                "code": 59651
            },
            "setIdx": 1,
            "setId": 1,
            "iconIdx": 3
        },
        {
            "icon": {
                "paths": [
                    "M896 256v-128h-896v704c0 35.346 28.654 64 64 64h864c53.022 0 96-42.978 96-96v-544h-128zM832 832h-768v-640h768v640zM128 320h640v64h-640zM512 448h256v64h-256zM512 576h256v64h-256zM512 704h192v64h-192zM128 448h320v320h-320z"
                ],
                "tags": ["newspaper", "news", "paper"],
                "defaultCode": 59652,
                "grid": 16,
                "attrs": []
            },
            "attrs": [],
            "properties": {
                "ligatures": "newspaper, news",
                "name": "newspaper",
                "order": 6,
                "id": 5,
                "prevSize": 32,
                "code": 59652
            },
            "setIdx": 1,
            "setId": 1,
            "iconIdx": 4
        },
        {
            "icon": {
                "paths": [
                    "M864 0c88.364 0 160 71.634 160 160 0 36.020-11.91 69.258-32 96l-64 64-224-224 64-64c26.742-20.090 59.978-32 96-32zM64 736l-64 288 288-64 592-592-224-224-592 592zM715.578 363.578l-448 448-55.156-55.156 448-448 55.156 55.156z"
                ],
                "tags": ["pencil", "write", "edit"],
                "defaultCode": 59653,
                "grid": 16,
                "attrs": []
            },
            "attrs": [],
            "properties": {
                "ligatures": "pencil, write",
                "name": "pencil",
                "order": 7,
                "id": 6,
                "prevSize": 32,
                "code": 59653
            },
            "setIdx": 1,
            "setId": 1,
            "iconIdx": 5
        },
        {
            "icon": {
                "paths": [
                    "M384 640l128-64 448-448-64-64-448 448-64 128zM289.3 867.098c-31.632-66.728-65.666-100.762-132.396-132.394l99.096-272.792 128-77.912 384-384h-192l-384 384-192 640 640-192 384-384v-192l-384 384-77.912 128z"
                ],
                "tags": ["pencil", "write", "edit"],
                "defaultCode": 59654,
                "grid": 16,
                "attrs": []
            },
            "attrs": [],
            "properties": {
                "ligatures": "pencil2, write2",
                "name": "pencil2",
                "order": 8,
                "id": 7,
                "prevSize": 32,
                "code": 59654
            },
            "setIdx": 1,
            "setId": 1,
            "iconIdx": 6
        }
    ],
    "height": 1024,
    "metadata": { "name": "icomoon" },
    "preferences": {
        "showGlyphs": true,
        "showQuickUse": true,
        "showQuickUse2": true,
        "showSVGs": true,
        "fontPref": {
            "prefix": "icon-",
            "metadata": { "fontFamily": "icomoon" },
            "metrics": { "emSize": 1024, "baseline": 6.25, "whitespace": 50 },
            "embed": false,
            "autoHost": true
        },
        "imagePref": {
            "prefix": "icon-",
            "png": true,
            "useClassSelector": true,
            "color": 0,
            "bgColor": 16777215,
            "classSelector": ".icon"
        },
        "historySize": 50,
        "showCodes": true,
        "gridSize": 16,
        "quickUsageToken": {
            "UntitledProject": "NWRmNjcxZTJiOGY2NjdmMjViNzdkOTliNWFiOWJhOGUjMSMxNjUzMDI0ODA4IyMjMTFiZDJhZjI4MDkz"
        }
    }
}
