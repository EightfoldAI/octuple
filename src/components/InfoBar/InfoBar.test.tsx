import React from 'react';
import Enzyme from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import MatchMediaMock from 'jest-matchmedia-mock';
import { InfoBar, InfoBarType } from './';
import { fireEvent, render } from '@testing-library/react';
import { IconName } from '../Icon';

Enzyme.configure({ adapter: new Adapter() });

let matchMedia: any;

describe('InfoBar', () => {
  beforeAll(() => {
    matchMedia = new MatchMediaMock();
  });

  afterEach(() => {
    matchMedia.clear();
  });

  test('Renders without crashing', () => {
    const { container, getByRole } = render(
      <InfoBar content={'InfoBar test render'} />
    );
    const infoBar = getByRole('alert');
    expect(() => container).not.toThrowError();
    expect(infoBar).toBeTruthy();
    expect(container).toMatchSnapshot();
  });

  test('Calls onClose callback when close button is clicked', () => {
    const onClose = jest.fn();
    const { container } = render(
      <InfoBar closable content={'InfoBar test onClose'} onClose={onClose} />
    );
    fireEvent.click(container.querySelector('.close-button'));
    expect(onClose).toHaveBeenCalled();
  });

  test('Renders a custom icon when the icon prop uses a custom icon', () => {
    const { container } = render(
      <InfoBar content={'InfoBar test icon'} icon={IconName.mdiAccountGroup} />
    );
    expect(container.querySelector('.icon')).toBeTruthy();
    expect(container).toMatchSnapshot();
  });

  test('InfoBar is Disruptive', () => {
    const { container } = render(
      <InfoBar
        content={'InfoBar test disruptive'}
        type={InfoBarType.disruptive}
      />
    );
    expect(container.querySelector('.disruptive')).toBeTruthy();
    expect(container).toMatchSnapshot();
  });

  test('InfoBar is Neutral', () => {
    const { container } = render(
      <InfoBar content={'InfoBar test neutral'} type={InfoBarType.neutral} />
    );
    expect(container.querySelector('.neutral')).toBeTruthy();
    expect(container).toMatchSnapshot();
  });

  test('InfoBar is Positive', () => {
    const { container } = render(
      <InfoBar content={'InfoBar test positive'} type={InfoBarType.positive} />
    );
    expect(container.querySelector('.positive')).toBeTruthy();
    expect(container).toMatchSnapshot();
  });

  test('InfoBar is Warning', () => {
    const { container } = render(
      <InfoBar content={'InfoBar test warning'} type={InfoBarType.warning} />
    );
    expect(container.querySelector('.warning')).toBeTruthy();
    expect(container).toMatchSnapshot();
  });

  test('InfoBar is bordered', () => {
    const { container } = render(
      <InfoBar bordered content={'InfoBar test border'} />
    );
    expect(container.querySelector('.bordered')).toBeTruthy();
    expect(container).toMatchSnapshot();
  });
});
