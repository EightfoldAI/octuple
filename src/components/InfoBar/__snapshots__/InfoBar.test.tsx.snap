// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoBar InfoBar is Disruptive 1`] = `
<div>
  <div
    class="info-bar disruptive"
    role="alert"
  >
    <span
      aria-hidden="false"
      class="icon icon-wrapper"
      role="presentation"
    >
      <svg
        role="presentation"
        style="width: 20px; height: 20px;"
        viewBox="0 0 24 24"
      >
        <path
          d="M13,9H11V7H13M13,17H11V11H13M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10,10 0 0,0 12,2Z"
          style="fill: currentColor;"
        />
      </svg>
    </span>
    <div
      class="content-wrapper"
    >
      <div
        class="message body2"
      >
        InfoBar test disruptive
      </div>
    </div>
  </div>
</div>
`;

exports[`InfoBar InfoBar is Neutral 1`] = `
<div>
  <div
    class="info-bar neutral"
    role="alert"
  >
    <span
      aria-hidden="false"
      class="icon icon-wrapper"
      role="presentation"
    >
      <svg
        role="presentation"
        style="width: 20px; height: 20px;"
        viewBox="0 0 24 24"
      >
        <path
          d="M13,9H11V7H13M13,17H11V11H13M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10,10 0 0,0 12,2Z"
          style="fill: currentColor;"
        />
      </svg>
    </span>
    <div
      class="content-wrapper"
    >
      <div
        class="message body2"
      >
        InfoBar test neutral
      </div>
    </div>
  </div>
</div>
`;

exports[`InfoBar InfoBar is Positive 1`] = `
<div>
  <div
    class="info-bar positive"
    role="alert"
  >
    <span
      aria-hidden="false"
      class="icon icon-wrapper"
      role="presentation"
    >
      <svg
        role="presentation"
        style="width: 20px; height: 20px;"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 2C6.5 2 2 6.5 2 12S6.5 22 12 22 22 17.5 22 12 17.5 2 12 2M10 17L5 12L6.41 10.59L10 14.17L17.59 6.58L19 8L10 17Z"
          style="fill: currentColor;"
        />
      </svg>
    </span>
    <div
      class="content-wrapper"
    >
      <div
        class="message body2"
      >
        InfoBar test positive
      </div>
    </div>
  </div>
</div>
`;

exports[`InfoBar InfoBar is Warning 1`] = `
<div>
  <div
    class="info-bar warning"
    role="alert"
  >
    <span
      aria-hidden="false"
      class="icon icon-wrapper"
      role="presentation"
    >
      <svg
        role="presentation"
        style="width: 20px; height: 20px;"
        viewBox="0 0 24 24"
      >
        <path
          d="M13 14H11V9H13M13 18H11V16H13M1 21H23L12 2L1 21Z"
          style="fill: currentColor;"
        />
      </svg>
    </span>
    <div
      class="content-wrapper"
    >
      <div
        class="message body2"
      >
        InfoBar test warning
      </div>
    </div>
  </div>
</div>
`;

exports[`InfoBar InfoBar is bordered 1`] = `
<div>
  <div
    class="info-bar bordered neutral"
    role="alert"
  >
    <span
      aria-hidden="false"
      class="icon icon-wrapper"
      role="presentation"
    >
      <svg
        role="presentation"
        style="width: 20px; height: 20px;"
        viewBox="0 0 24 24"
      >
        <path
          d="M13,9H11V7H13M13,17H11V11H13M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10,10 0 0,0 12,2Z"
          style="fill: currentColor;"
        />
      </svg>
    </span>
    <div
      class="content-wrapper"
    >
      <div
        class="message body2"
      >
        InfoBar test border
      </div>
    </div>
  </div>
</div>
`;

exports[`InfoBar Renders a custom icon when the icon prop uses a custom icon 1`] = `
<div>
  <div
    class="info-bar neutral"
    role="alert"
  >
    <span
      aria-hidden="false"
      class="icon icon-wrapper"
      role="presentation"
    >
      <svg
        role="presentation"
        style="width: 20px; height: 20px;"
        viewBox="0 0 24 24"
      >
        <path
          d="M12,5.5A3.5,3.5 0 0,1 15.5,9A3.5,3.5 0 0,1 12,12.5A3.5,3.5 0 0,1 8.5,9A3.5,3.5 0 0,1 12,5.5M5,8C5.56,8 6.08,8.15 6.53,8.42C6.38,9.85 6.8,11.27 7.66,12.38C7.16,13.34 6.16,14 5,14A3,3 0 0,1 2,11A3,3 0 0,1 5,8M19,8A3,3 0 0,1 22,11A3,3 0 0,1 19,14C17.84,14 16.84,13.34 16.34,12.38C17.2,11.27 17.62,9.85 17.47,8.42C17.92,8.15 18.44,8 19,8M5.5,18.25C5.5,16.18 8.41,14.5 12,14.5C15.59,14.5 18.5,16.18 18.5,18.25V20H5.5V18.25M0,20V18.5C0,17.11 1.89,15.94 4.45,15.6C3.86,16.28 3.5,17.22 3.5,18.25V20H0M24,20H20.5V18.25C20.5,17.22 20.14,16.28 19.55,15.6C22.11,15.94 24,17.11 24,18.5V20Z"
          style="fill: currentColor;"
        />
      </svg>
    </span>
    <div
      class="content-wrapper"
    >
      <div
        class="message body2"
      >
        InfoBar test icon
      </div>
    </div>
  </div>
</div>
`;

exports[`InfoBar Renders without crashing 1`] = `
<div>
  <div
    class="info-bar neutral"
    role="alert"
  >
    <span
      aria-hidden="false"
      class="icon icon-wrapper"
      role="presentation"
    >
      <svg
        role="presentation"
        style="width: 20px; height: 20px;"
        viewBox="0 0 24 24"
      >
        <path
          d="M13,9H11V7H13M13,17H11V11H13M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10,10 0 0,0 12,2Z"
          style="fill: currentColor;"
        />
      </svg>
    </span>
    <div
      class="content-wrapper"
    >
      <div
        class="message body2"
      >
        InfoBar test render
      </div>
    </div>
  </div>
</div>
`;
